docs/pages/cairo/enum.md

```diff
@@ -1,21 +1,69 @@
 ## Enum
 
-Enums are very useful in game design, as they simplify the creation of clean, complex logic.
+## What is Enum
+Enums, short for "enumerations," are a way to define a custom data type that consists of a fixed set of named values, called variants. Enums are useful for representing a collection of related values where each value is distinct and has a specific meaning. Enums are particularly useful in game development for representing game states, player actions, or any other set of related constants that a game might need to track.
+```rust
+# [derive(Serde, Copy, Drop, Introspect, PartialEq, Print)]
+enum PlayerCharacter {
+    Godzilla,
+    Dragon,
+    Fox,
+    Rhyno
+}
 
-You can define an enum as follows:
+```
+In this example, we've defined an enum called PlayerCharacter with four variants: `Godzilla`, `Dragon`, `Fox`, and `Rhyno`. The naming convention is to use PascalCase for enum variants. Each variant represents a distinct value of the PlayerCharacter type. In this particular example, variants don't have any associated value. 
+Now let's imagine that our variants have associated values, We can define a new PlayerCharacter enum:
 
 ```rust
+# [derive(Serde, Copy, Drop, Introspect, PartialEq, Print)]
+enum PlayerCharacter {
+    Godzilla :u128,
+    Dragon : u128,
+    Fox : u128,
+    Rhyno : u128
+}
+
+```
+
+In the next example, we will see that it is also possible to associate different data types with each variant. The PlayerCharacter enum we will define serves as a blueprint for creating various types of characters, each with their unique attributes. This design choice is particularly useful in game development, where you often need to manage different types of entities (like characters, items, or enemies) that share some common properties but also have unique characteristics.
+
+``` rust 
+# [derive(Serde, Copy, Drop, Introspect, PartialEq, Print)]
+enum PlayerCharacter {
+    Godzilla ,
+    Fox  : felt252,
+}
+
+    // Instantiate some characters with associated values
+let godzilla = PlayerCharacter::Godzilla;
+let fox = PlayerCharacter::Fox(90);
+```
```

Comment:
This is not supported in Dojo yet. All enum variants must have the same type for now.